scripts/homepage.js

```diff
@@ -227,19 +228,22 @@ $(function () {
     // Change Button text on payment click
     function updateDownloadButton () {
         var payment_amount = $('#amount-custom').val() * 100;
-        if (
-            $('input#amount-custom').hasClass('checked') &&
-            payment_amount < payment_minimum &&
-            $('#amount-custom').val() != ''
-        ) {
+        if ($('#amounts').children().length <= 1) {
             $('#download').text('Download elementary OS');
         } else {
-            $('#download').text('Purchase elementary OS');
+            if (($('input.button#amount-custom').hasClass('checked') &&
+                 $('input.button#amount-custom').val() >= 1) || (
```

Comment:
The custom amount field will never be checked while having an amount smaller than the minimum. You also seem to have lost the `payment_amount < payment_minimum`. I wouldn't worry about the button class, it's not relevant programatically, it's just a css target.